src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -757,12 +770,25 @@ bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invari
   Node* offset = NULL;
   jlong scale = 0;
   Node* iv = _head->as_BaseCountedLoop()->phi();
+  Compile* C = Compile::current();
+  const uint old_unique_idx = C->unique();
   if (is_range_check_if(iff, phase, T_INT, iv, range, offset, scale)) {
     if (!invar.is_invariant(range)) {
```

Comment:
First, the fix is reasonable to me.

My only complain is original code flow. If we do `return false;` for each check we should the same for `is_range_check_if()` check instead of returning `false` at the very end you lost logic why it is `false`.

Reply:
Thanks for your review Vladimir! Good point, I changed that into a direct bailout.